### ISSUE TITLE
[Performance] Static assets are not compressed in production

### ISSUE THREAD

**Issue by janicduplessis** [Expensify/App#49191]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## What performance issue do we need to solve?

When loading newdot in production, the static text assets (js, css, html, json, etc.) are not compressed. This leads to long loading time, expecially on slower networks.

For example when loading https://new.expensify.com we can observe in Safari devtools that no compression is done and the content is served as text/javascript for https://new.expensify.com/main-5c8169422a350d70ff13.bundle.js.

<img width="1096" alt="image" src="https://github.com/user-attachments/assets/dbf98249-1808-40bd-bc2a-b569f92561b2">

We can compare this with a request to a 3rd party library like appleid and see what it looks like when compressed.

<img width="1092" alt="image" src="https://github.com/user-attachments/assets/771b2fb7-3c91-4603-8ee0-f331fa1edf4a">

If I compress this JS bundle file manually the size goes from 4.4mb to 1.2mb.

## What is the impact of this on end-users?

App is slow to load initially on slower connections.

## List any benchmarks that show the severity of the issue
Please also provide exact steps taken to collect metrics above if any so we can independently verify the results.

## Proposed solution (if any)

Enable compression of static assets.

## List any benchmarks after implementing the changes to show impacts of the proposed solution (if any)

Collect network traces in Safari or Chrome devtools and check the size of various files.

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [x] Android: mWeb Chrome
- [ ] iOS: Native
- [x] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [x] MacOS: Desktop

**Version Number:** 9.0.33-4
**Reproducible in staging?:** Yes
**Reproducible in production?:** Yes
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation
**Expensify/Expensify Issue URL:**
**Issue reported by:** @janicduplessis 
**Slack conversation:** https://expensify.slack.com/archives/C05LX9D6E07/p1726259564418389

[View all open jobs on Upwork](https://www.upwork.com/ab/jobs/search/?q=Expensify%20React%20Native&sort=recency&user_location_match=2)

**Comment by justinpersaud** [comment]
Will look at soon.

**Comment by justinpersaud** [comment]
![image](https://github.com/user-attachments/assets/a9754272-284c-4320-bc61-4a84eb4c5fb5)

**Comment by justinpersaud** [comment]
It's not obvious to me why these aren't being served compressed so I opened a cloudflare ticket.

**Comment by justinpersaud** [comment]
Having trouble getting a response from Cloudflare through our support ticket on this. I pinged our account team on this.

**Comment by justinpersaud** [comment]
Got a response from Cloudflare but still have not identified where the issue is. Still working through it.

**Comment by justinpersaud** [comment]
Ok, I think we have figured this one out.

We had a rule enabled from awhile back when we had problems with videos not playing back in Safari properly. It ended up having Cloudflare serving strong etags on cached files, which in turn, disabled compression. We also had another rule that used strong etags for mp4 files which were affected by the issue, so we didn't actually need strong etags everywhere.

I turned it off and it looks like it is working now.

```
➜ curl -s -I -H "Accept-Encoding: gzip" https://staging.new.expensify.com/main-dbcc01d6282a3006774f.bundle.js | grep content-encoding
content-encoding: gzip
```
